src/map.rs

```diff
@@ -449,9 +451,104 @@ where
     }
 }
 
+impl<'a, K, V, S> ParallelExtend<Arc<Element<K, V>>> for &'a ConMap<K, V, S>
+where
+    K: Hash + Eq + Send + Sync,
+    V: ?Sized + Send + Sync,
+    S: BuildHasher + Sync,
+{
+    fn par_extend<T>(&mut self, par_iter: T)
+    where
+        T: IntoParallelIterator<Item = Arc<Element<K, V>>>,
+    {
+        par_iter.into_par_iter().for_each(|n| {
+            self.insert_element(n);
+        });
+    }
+}
+
+impl<K, V, S> ParallelExtend<(K, V)> for ConMap<K, V, S>
+where
+    K: Hash + Eq + Send + Sync,
+    S: BuildHasher + Sync,
+    V: Send + Sync,
+{
+    fn par_extend<T>(&mut self, par_iter: T)
+    where
+        T: IntoParallelIterator<Item = (K, V)>,
+    {
+        par_iter.into_par_iter().for_each(|(k, v)| {
+            self.insert(k, v);
+        });
```

Comment:
Would it be possible to for 3 of the implementations to delegate into the 4th, like the ordinary Extends do? I'd prefer to have just one copy of the „real“ implementation even though it is really small and trivial, so if there's ever a bug (unlikely, but principles…), there's only one place to fix.